crates/ethereum/node/tests/e2e/p2p.rs

```diff
@@ -45,3 +57,115 @@ async fn can_sync() -> eyre::Result<()> {
 
     Ok(())
 }
+
+#[tokio::test]
+async fn e2e_test_send_transactions() -> eyre::Result<()> {
+    reth_tracing::init_test_tracing();
+
+    let seed: [u8; 32] = rand::thread_rng().gen();
+    let mut rng = StdRng::from_seed(seed);
+    println!("Seed: {:?}", seed);
+
+    let chain_spec = Arc::new(
+        ChainSpecBuilder::default()
+            .chain(MAINNET.chain)
+            .genesis(serde_json::from_str(include_str!("../assets/genesis.json")).unwrap())
+            .cancun_activated()
+            .prague_activated()
+            .build(),
+    );
+
+    let (mut nodes, _tasks, wallet) =
+        setup_engine::<EthereumNode>(2, chain_spec.clone(), false).await?;
+    let mut node = nodes.pop().unwrap();
+    let signers = wallet.gen();
+    let provider = ProviderBuilder::new().with_recommended_fillers().on_http(node.rpc_url());
+
+    // simple contract which writes to storage on any call
+    let dummy_bytecode = bytes!("6080604052348015600f57600080fd5b50602880601d6000396000f3fe4360a09081523360c0526040608081905260e08152902080805500fea164736f6c6343000810000a");
+    let mut call_destinations = signers.iter().map(|s| s.address()).collect::<Vec<_>>();
+
+    // Produce 100 random blocks with random transactions
+    for _ in 0..100 {
+        let tx_count = rng.gen_range(1..20);
+
+        let mut pending = vec![];
+        for _ in 0..tx_count {
+            let signer = signers.choose(&mut rng).unwrap();
+            let tx_type = TxType::try_from(rng.gen_range(0..=4)).unwrap();
+
+            let mut tx = TransactionRequest::default().with_from(signer.address());
+
+            let should_create =
+                rng.gen::<bool>() && tx_type != TxType::Eip4844 && tx_type != TxType::Eip7702;
+            if should_create {
+                tx = tx.into_create().with_input(dummy_bytecode.clone());
+            } else {
+                tx = tx.with_to(*call_destinations.choose(&mut rng).unwrap()).with_input(
+                    (0..rng.gen_range(0..10000)).map(|_| rng.gen()).collect::<Vec<u8>>(),
+                );
+            }
+
+            if matches!(tx_type, TxType::Legacy | TxType::Eip2930) {
+                tx = tx.with_gas_price(provider.get_gas_price().await?);
+            }
+
+            if rng.gen::<bool>() || tx_type == TxType::Eip2930 {
+                tx = tx.with_access_list(
+                    vec![AccessListItem {
+                        address: *call_destinations.choose(&mut rng).unwrap(),
+                        storage_keys: (0..rng.gen_range(0..100)).map(|_| rng.gen()).collect(),
+                    }]
+                    .into(),
+                );
+            }
+
+            if tx_type == TxType::Eip7702 {
+                let signer = signers.choose(&mut rng).unwrap();
+                let auth = Authorization {
+                    chain_id: provider.get_chain_id().await?,
+                    address: *call_destinations.choose(&mut rng).unwrap(),
```

Comment:
this will never actually delegate as it will try to delegate to an empty account